crud/common/utils.lua

```diff
@@ -282,11 +285,57 @@ function utils.is_uuid(value)
     return ffi.istype(uuid_t, value)
 end
 
-function utils.format_result(rows, space)
-    return {
-        metadata = table.copy(space:format()),
-        rows = rows,
-    }
+local function get_field_format(space_format, field_name)
+    dev_checks('table', 'string')
+
+    local metadata = space_format_cache[space_format]
+    if metadata ~= nil then
+        return metadata[field_name]
+    end
+
+    space_format_cache[space_format] = {}
+    for _, field in ipairs(space_format) do
+        space_format_cache[space_format][field.name] = field
+    end
+
+    return space_format_cache[space_format][field_name]
+end
+
+local function filter_format_fields(space_format, field_names)
+    dev_checks('table', 'table')
+
+    local metadata = {}
+
+    for i, field in ipairs(field_names) do
+        metadata[i] = get_field_format(space_format, field)
+        if metadata[i] == nil then
+            return nil, FilterFieldsError:new(
+                    'Space format doesn\'t contain field named %q', field
+            )
+        end
+    end
+
+    return metadata
+end
+
+function utils.format_result(rows, space, field_names)
+    local result = {}
+    local err
+    local space_format = table.copy(space:format())
```

Comment:
table.copy here is an attempt to protect space:format() value form user modifications.

If you need to create custom metadata you actually perform this copy manually. So you don't need to perform `table.copy` if fields are specified.